src/builtin_set.cpp

```diff
@@ -805,7 +805,6 @@ static int builtin_set_set(const wchar_t *cmd, set_cmd_opts_t &opts, int argc, w
 
 /// The set builtin creates, updates, and erases (removes, deletes) variables.
 maybe_t<int> builtin_set(parser_t &parser, io_streams_t &streams, wchar_t **argv) {
-    const int incoming_exit_status = parser.get_last_status();
     wchar_t *cmd = argv[0];
```

Comment:
I think this commit breaks the default git prompt, not sure why; it no longer recognises that I'm on `master`.

```sh
fish-shell (master)> HOME=$PWD ./build/fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
~ ((cfa92bba…))>
```

Reply:
Yeah... `if not set branch (command git symbolic-ref HEAD 2>/dev/null)`. Looks like the `not` is being ignored there somehow. I will fix that and add a test for it.

Reply:
Thanks for catching this! I've added logic in exec.cpp to handle negation when no status is returned, and added some test cases to cover this.